lib/streamlit/elements/layouts.py

```diff
@@ -402,11 +406,64 @@ def expander(self, label: str, expanded: bool = False) -> "DeltaGenerator":
         expandable_proto.label = label
 
         block_proto = BlockProto()
-        block_proto.allow_empty = True
+        block_proto.allow_empty = False
         block_proto.expandable.CopyFrom(expandable_proto)
 
         return self.dg._block(block_proto=block_proto)
 
+    @gather_metrics("status")
+    def status(
+        self,
+        label: str,
+        *,
+        expanded: bool = False,
+        state: Literal["running", "complete", "error"] = "running",
+    ) -> "StatusContainer":
+        """Insert a status container.
```

Comment:
*Just rough draft ideas, feel free to modify:*

Insert a status container.

Inserts a container into your app that is typically used to show the status and details of a process or task. The container can hold multiple elements and can be expanded or collapsed by the user similar to st.expander. When collapsed, all that is visible is the status icon and label.

The label, state, and expanded state can all be updated by calling "update()" on the returned object. To add elements to the returned container, you can use "with" notation (preferred) or just call methods directly on the returned object.

By default, st.status() initializes in the "running" state. When called using "with" notation, it automatically updates to the "complete" state at the end of the "with" block. <ideally add an example>

Reply:
Update the text 👍